content/en/blog/_posts/2024-04-23-recursive-read-only-mounts.md

```diff
@@ -0,0 +1,97 @@
+---
+layout: blog
+title: 'Kubernetes 1.30: Read-only volume mounts can be finally literally read-only'
+date: 2024-04-23
+slug: recursive-read-only-mounts
+---
+
+**Author:** Akihiro Suda (NTT)
+
+Kubernetes 1.30 added alpha support for "recursive read-only mounts".
+
+Prior to this release, read-only volume mounts were surprisingly not really read-only under specific conditions.
+
+### Read-only volume mounts are not really read-only
+
+Volume mounts are widely misunderstood.
+
+The following manifest is expected to make everything under `/mnt` in the containers read-only:
+```yaml
+apiVersion: v1
+kind: Pod
+spec:
+  volumes:
+    - name: mnt
+      hostPath:
+        path: /mnt
+  containers:
+    - volumeMounts:
+        - name: mnt
+          mountPath: /mnt
+          readOnly: true
+```
+
+However, it actually leaves submounts beneath `/mnt` (e.g., NFS or removable media) writable.
+
+### New mount option: recursiveReadOnly
+
+Kubernetes 1.30 added a new mount option `recursiveReadOnly` so as to make submounts recursively read-only.
+
+The option can be enabled as follows:
+```yaml
+apiVersion: v1
+kind: Pod
+spec:
+  volumes:
+    - name: mnt
+      hostPath:
+        path: /mnt
+  containers:
+    - volumeMounts:
+        - name: mnt
+          mountPath: /mnt
+          readOnly: true
+          # NEW
+          # Possible values are `Enabled`, `IfPossible`, and `Disabled`.
+          # Needs to be specified in conjunction with `readOnly: true`.
+          recursiveReadOnly: Enabled
```

Comment:
You can use the `highlight` shortcode for emphasis. See https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/#cleanup-all-secrets-encrypted (and its source) for an example.

Reply:
done